### PR TITLE
fixes #23612 - robottelo test port content credentials (gpg_keys)

### PR DESCRIPTION
**Pull request by pondrejk**
Port robottelo tier1 api tests for GPG Key 
part of robottelo minitest port project: https://github.com/SatelliteQE/robottelo/projects/1

Related Robottelo issue: SatelliteQE/robottelo#5870

```
rake test:katello TEST=../katello/test/controllers/api/v2/content_credentials_controller_test.rb
[snip]

# Running:

..........

Finished in 4.736510s, 2.1113 runs/s, 5.7004 assertions/s.
10 runs, 27 assertions, 0 failures, 0 errors, 0 skips

```

### CODE REVIEW

File: test/models/gpg_keys_test.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require 'katello_test_helper'
+
+module Katello
+  class GpgKeyTest < ActiveSupport::TestCase
+    should allow_values(*valid_name_list).for(:name)
+    should_not allow_values(*invalid_name_list).for(:name)
+    should_not allow_values(['', ' ', "\t", *RFauxFactory.gen_strings(247).values]).for(:content)
```
Comment (ldjebran) [inline test/models/gpg_keys_test.rb]: this should be rewritten to (as the current implementation will be interpreted as one  array value, and will not be allowed any way):
```ruby
should_not allow_values('', ' ', "\t", *RFauxFactory.gen_strings(247).values).for(:content)
```

File: test/controllers/api/v2/content_credentials_controller_test.rb [hunk 2]
```diff
@@ -69,5 +69,75 @@ def test_cert_content
       assert_response :success, @response.body
       assert_equal content, @cert.reload.content
     end
+
+    def test_create
```
Comment (beav) [inline test/controllers/api/v2/content_credentials_controller_test.rb]: :+1:

File: test/controllers/api/v2/content_credentials_controller_test.rb [hunk 3]
```diff
@@ -69,5 +69,75 @@ def test_cert_content
...
+      content = "abc123"
+      temp_content_file = Tempfile.new(content)
+      temp_content_file.write(content)
+      temp_content_file.rewind
+      uploaded_file = Rack::Test::UploadedFile.new(temp_content_file.path, "text/plain")
+      post :create, params: { :organization_id => @organization.id, :name => test_key_name, :content_type => "gpg_key", :content => uploaded_file}
+      assert_response :success
+    end
+
+    def test_create_existing_name
```
Comment (beav) [inline test/controllers/api/v2/content_credentials_controller_test.rb]: :+1:

File: test/controllers/api/v2/content_credentials_controller_test.rb [hunk 4]
```diff
@@ -69,5 +69,75 @@ def test_cert_content
...
+      temp_content_file.write(content)
+      temp_content_file.rewind
+      uploaded_file = Rack::Test::UploadedFile.new(temp_content_file.path, "text/plain")
+      post :create, params: { :organization_id => @organization.id, :name => @gpg_key.name, :content_type => "gpg_key", :content => uploaded_file }
+      assert_response 422, @response.body
+      response = JSON.parse(@response.body)
+      assert_equal response["displayMessage"], "Validation failed: Name has already been taken"
+    end
+
+    def test_create_invalid_content
```
Comment (beav) [inline test/controllers/api/v2/content_credentials_controller_test.rb]: this is covered by https://github.com/Katello/katello/blob/master/spec/models/gpg_key_spec.rb#L32

**Comment by beav** [comment]
It looks like a lot of the tests are already in https://github.com/Katello/katello/blob/master/spec/models/gpg_key_spec.rb, I didn't notice until after I started reviewing. Can the additional tests be moved over there?

**Comment by jlsherrill** [comment]
@pondrejk this looks like its waiting on you

**Comment by jturel** [comment]
[test katello]

**Comment by akofink** [comment]
Closing due to inactivity.

**Pull request closed without merging**